### DISCUSSION TITLE
Custom search suggestion HTML support

### DISCUSSION THREAD

**Discussion by SinghGurd** [microsoft-search/pnp-modern-search#2216] [General]
Can we use HTML for a custom search suggestion e.g., the following suggestion? This works Ok for display name but not for description. Another problem is when the suggestion is selected entire HTML appears in the search box. I am trying to use the search box suggestion flyout itself to show the most important pieces of a profile on our Intranet home page without using the search results webpart. 

Alternatively, is there an event available that I could use to change the displayText back to plain-text on suggestion selection. I have tried the _onSuggestionSelected to reset displayText="Bob Smith" but it has no impact on text displayed inside search box.

`result = { displayText: "<div>Bob Smith<br>Consultant<br>Engineering</div>",
            description: "<div>Bob Smith<br>Consultant<br>Engineering</div>",
            groupName: '',
            iconSrc: "",
            onSuggestionSelected: this._onSuggestionSelected,
            hoverText: ``
          };`

**Comment by FranckyC** [comment]
The current implementation only supports HTML for display text, not the description, so this is a limitation. For the selected text we could add a new property on the `ISuggestion` interface to set the selected text by yourself. Would that make sense?

I mark the discussion as an enhancement.

Thanks!

**Reply by SinghGurd** [reply]
Great...Thank you for the response @FranckyC
